HousingManagementSystemApi/UseCases/VerifyPropertyEligibilityUseCase.cs

```diff
@@ -0,0 +1,88 @@
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using HACT.Dtos;
+using HousingManagementSystemApi.Gateways;
+
+namespace HousingManagementSystemApi.UseCases
+{
+    using System.Linq;
+    using System.Reflection.Metadata.Ecma335;
+    using Amazon.Runtime.Internal.Util;
+    using Hackney.Shared.Asset.Domain;
+    using Hackney.Shared.Tenure.Domain;
+    using HousingManagementSystemApi.Models;
+    using Microsoft.Extensions.Logging;
+
+    public class VerifyPropertyEligibilityUseCase : IVerifyPropertyEligibilityUseCase
+    {
+        private readonly IAssetGateway _assetGateway;
+        private readonly IEnumerable<AssetType> _assetTypes;
+        private readonly ITenureGateway _tenureGateway;
+        private readonly ILogger<VerifyPropertyEligibilityUseCase> _logger;
+
+        private IEnumerable<string> EligibleTenureCodes { get; }
+
+        public VerifyPropertyEligibilityUseCase(
+            IAssetGateway assetGateway,
+            IEnumerable<AssetType> assetTypes,
+            ITenureGateway tenureGateway,
+            IEnumerable<TenureType> eligibleTenureTypes,
+            ILogger<VerifyPropertyEligibilityUseCase> logger)
+        {
+            _assetGateway = assetGateway;
+            _assetTypes = assetTypes;
+            _tenureGateway = tenureGateway;
+            EligibleTenureCodes = eligibleTenureTypes.Select(x => x.Code);
+            _logger = logger;
+        }
+
+        public async Task<PropertyEligibilityResult> Execute(string propertyId)
+        {
+            _logger.LogInformation("Calling VerifyPropertyEligibilityUseCase for {PropertyId}", propertyId);
+
+            if (string.IsNullOrEmpty(propertyId))
+            {
+                _logger.LogInformation("The property ID {PropertyId} was null or empty. Throwing Exception", propertyId);
+                throw new ArgumentNullException(nameof(propertyId));
+            }

```

Comment:
Would it make more sense for this logic to be in the controller?

Reply:
Promoting true component re-use, this could be used by something other than that controller. So, it should be in both. It's necessary for this use case to work.